deployment/kubernetes.coffee

```diff
@@ -0,0 +1,154 @@
+fs            = require 'fs'
+{ isAllowed } = require './grouptoenvmapping'
+
+generatePodDef = ->
+
+  '''
+  apiVersion: v1
+  kind: Pod
+  metadata:
+    name: backend
+    namespace: koding
+  spec:
+    restartPolicy: Never
+    containers:
+  '''
+
+generateContainerSection = (app, options = {}) ->
+
+  container =
+    name          : app
+    image         : options.kubernetes.image
+    workingDir    : '/opt/koding'
+    command       : options.kubernetes.command
+    env           : generateEnvvarsSection options.kubernetes.envVariables?, options
+    ports					: generatePortsSection options.kubernetes.ports?, options
+    mounts        : generateMountSection options.kubernetes.mounts?, options
+
+  containerSection =
+  """
+  \n    - name: #{container.name}
```

Comment:
You dont need to work with strings. First  generate an object then use a yaml generator to convert that to yaml.